base/doc/clsguide.tex

```diff
@@ -1138,6 +1138,40 @@ \subsection{Normalising spacing}
 and so need not be explicitly set; however, if it is explicitly made non-empty
 in a class file then automatic default setting will be over-ridden.
 
+\subsection{Querying localisation}
+
+Localisation information is needed to customise a range of outputs. The
+\LaTeX{} kernel does not itself manage localisation, which is well-served
+by the bundles \pkg{babel} and \pkg{polyglossia}. To allow the kernel and
+other packages to access the current localisation information provided
+by \pkg{babel} or \pkg{polyglossia}, the command \cs{BCPdata} is defined
+by the kernel. The initial kernel definition absorbs one argument and
+expands to an empty value, as the kernel does not track localisation.
+However, if \pkg{babel} or \pkg{polyglossia} are loaded, it is redefined
+to takes one argument and expands to the BCP-47 information matching this
+argument. The supported arguments are the BCP-47 tag breakdowns:
+\begin{itemize}
+  \item \texttt{language} (e.g.,~\texttt{de})
+  \item \texttt{region} (e.g.,~\texttt{AT})
+  \item \texttt{script} (e.g.,~\texttt{Latn})
+  \item \texttt{variant} (e.g.,~\texttt{1901})
+  \item \texttt{extension.t} (transformation, e.g.,~\texttt{en-t-ja})
+  \item \texttt{extension.u} (additional locale
+    information, e.g.,~\texttt{ar-u-nu-latn})
+  \item \texttt{extension.x} (private use area, e.g.,~\texttt{la-x-classic})
+\end{itemize}
+The information for the \emph{main} language for a document is be provided
+if these are prefixed by \texttt{main.}, e.g.~\texttt{main.language} will
+expand to the main language even if another language is currently active.
+
+In addition to the tag breakdown, the following semantic arguments are
+supported
+\begin{itemize}
+  \item \texttt{casing} The tag for case changing, e.g.~\texttt{el-x-iota}
+    could be selected rather than \texttt{el} to select a capital adscript
+    iota on uppercasing an \emph{ypogegrammeni}
+\end{itemize}
+
```

Comment:
this section seems a bit abstract, would probably help to give some example of a package-level command definition that uses this, assuming some minimal  babel setup.